src/main/java/LinkedQueue.java

```diff
@@ -0,0 +1,125 @@
+import java.util.NoSuchElementException;
+
+public class LinkedQueue<T> implements Queue<T> {
+
+    private Node<T> front;
+    private Node<T> rear;
+    private int size;
+
+    public LinkedQueue() {
+        front = null;
+        rear = null;
+        size = 0;
+    }
+
+    @Override
+    public void enqueue(T element) {
+        Node<T> toEnqueue = new Node<>(element);
+
+        // If nothing is in the queue, then the front is null
+        // and the enqueued element will become the front,
+        // otherwise, the enqueued element is added after the
+        // current rear.
+        if (isEmpty()) {
+            front = toEnqueue;
+        }
+        else {
```

Comment:
Just fyi (and #56), I've only seen this `if/else` style from "non professional" code, for lack of a better word. You're way more likely to see the `else` inline with the `}`.

```java
if (condition) {
} else {
}
```